### DISCUSSION TITLE
New JSON feature

### DISCUSSION THREAD

**Discussion by TheYellowArchitect** [peter-kish/gloot#243] [Ideas]
Just wanted to share this PR from Godot 4.4 dev2 ["Ability to convert native engine types to JSON and back"](https://github.com/godotengine/godot/pull/92656) 
since a big rewrite is happening for gloot, and JSON is important

**Comment by peter-kish** [comment]
Interesting. Looks like a very useful feature for serializing/deserializing data.
At the same time, I'm not sure how I feel about it when it comes to user-provided JSON files (i.e. protosets).

We would basically expect the user to write:
```gdscript
"size": { 
  "__gtype":"Vector2",
  "values":[1,1]
},
```
instead of :
```gdscript
"size": "Vector2i(1, 1)"
```
which seems a bit cumbersome.

**Reply by TheYellowArchitect** [reply]
For that code example, expanding the GUI Editor Tool would work

**Reply by peter-kish** [reply]
Yeah, if the editor was the only way of creating protosets, that would solve the issue (it wouldn't matter how the data is stored "uner the hood"). But I also want to support writing protosets as JSON and I would like the notation to be as simple as possible.
In fact, version 3 will only support protosets as raw JSON and the reason for that is to simplify the addon by eliminating the editor code. That way protoset editors can be implemented as separate addons, or even external tools.
